backend/src/main/java/atlantbh/restaurants/models/filters/RestaurantFilterBuilder.java

```diff
@@ -0,0 +1,118 @@
+package atlantbh.restaurants.models.filters;
+
+import atlantbh.restaurants.models.sortkeys.RestaurantSortKeys;
+import org.hibernate.Criteria;
+import org.hibernate.criterion.Criterion;
+import org.hibernate.criterion.MatchMode;
+import org.hibernate.criterion.Restrictions;
+
+public class RestaurantFilterBuilder extends BaseFilterBuilder<RestaurantSortKeys, RestaurantFilterBuilder> {
+
+    private String query;
+    private String name;
+    private Integer priceRange;
+    private String location;
+    private String category;
+    private String cousine;
+    //TODO implement filtering for average rating of restaurant
+
+    public RestaurantFilterBuilder() {
+        query = null;
+        name = null;
+        priceRange = null;
+        location = null;
+        category = null;
+        cousine = null;
+    }
+
+    @Override
+    public Criteria addConditions(Criteria rootCriteria, boolean isCountCriteria) {
+
+        if (query != null && query.trim().length() > 0) { // ignore whtiespace characters
```

Comment:
We should discuss these conditions tomorrow.. this is not right place to trim strings.. it should be done on frontend, and some generic method, but not in filter builder. Also you can use org.apache.commons.lang lib for example
`StringUtils.isNotBlank(value);` etc..

Reply:
Fixed in 8c841a20e489a31c66d177c8682339f9d4f19de0